src/libs/Navigation/AppNavigator/WorkspaceSettingsDrawerNavigator.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import {createDrawerNavigator} from '@react-navigation/drawer';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import styles, {getNavigationDrawerStyle, getWorkspaceSettingsDrawerType} from '../../../styles/styles';
+import WorkspaceCard from '../../../pages/workspace/WorkspaceCard';
+import WorkspaceSidebar from '../../../pages/workspace/WorkspaceSidebar';
+
+const propTypes = {
+    ...windowDimensionsPropTypes,
+};
+const defaultProps = {};
+const Drawer = createDrawerNavigator();
+
+const WorkspaceSettingsDrawerNavigator = props => (
+    <Drawer.Navigator
+        defaultStatus={props.isSmallScreenWidth ? 'open' : 'closed'}
+        sceneContainerStyle={styles.navigationSceneContainer}
+        drawerContent={() => <WorkspaceSidebar />}
+        screenOptions={{
+            cardStyle: styles.navigationScreenCardStyle,
+            headerShown: false,
+            drawerType: getWorkspaceSettingsDrawerType(props.isSmallScreenWidth),
+            drawerStyle: getNavigationDrawerStyle(
+                props.windowWidth,
+                props.windowHeight,
+                props.isSmallScreenWidth,
+            ),
+            swipeEdgeWidth: 500,
```

Comment:
NAB, but now that we have a second drawer navigator some of these options could maybe be moved so they can be shared between the two navigators. This might make migrating things easier when react-navigation changes something. One idea could be to create a helper function that returns this function (but with the shared options).

Reply:
Yeah. I was planning to create a helper as we did for stack navigators which takes options that override the passed default props here.